src/bots/DepositBot.js

```diff
@@ -0,0 +1,272 @@
+    this._lastError = null
+
+    this._botInfo = 'DepositBot - v' + getVersion()
+  }
+
+  async _doStart () {
+    logger.debug({ msg: 'Initialized bot: ' + this.name })
+
+    // Check if the bots need to deposit periodically
+    this._depositFunds()
+    setInterval(() => {
+      return this._depositFunds()
+    }, this._checkTimeInMilliseconds)
+  }
+
+  async _doStop () {
+    logger.debug({ msg: 'Bot stopped: ' + this.name })
+  }
+
+  async _depositFunds () {
+    this._lastCheck = new Date()
+    let botHasDepositedFunds
+
+    function isInactiveTimeLapse ({ from, to }) {
+      const date1 = dateUtil.newDate(from, 'HH:mm')
+      const date2 = dateUtil.newDate(to, 'HH:mm')
+      return dateUtil.nowIsBetween(date1, date2)
+    }
+
+    const isWaitingTime = this._inactiveTimeLapse.some(isInactiveTimeLapse)
+    if (isWaitingTime) {
+      // We stop deposit funds execution
+      logger.debug('We are at an inactive time lapse, claim your funds now')
+      return false
+    }
+
+    try {
+      logger.debug('Tokens by account: %O', this._tokensByAccount)
+      const accountKeys = Object.keys(this._tokensByAccount)
+
+      const accountAddressesPromises = accountKeys.map(accountKey => {
+        return getBotAddress(this._ethereumClient, accountKey)
+      })
+
+      const accountAddresses = await Promise.all(accountAddressesPromises)
+      logger.debug('Account addresses: %O', accountAddresses)
+
+      const balanceOfEtherPromises = accountAddresses.map(account => {
+        // Get ETH balance
+        return this._dxInfoService.getBalanceOfEther({ account })
+      })
+
+      const balanceOfTokensPromises = accountAddresses.map((account, index) => {
+        // Get balance of ERC20 tokens
+        return this._dxInfoService.getAccountBalancesForTokensNotDeposited({
+          tokens: this._tokensByAccount[accountKeys[index]].tokens,
+          account
+        })
+      })
+
+      const balancesOfEther = await Promise.all(balanceOfEtherPromises)
+      logger.debug('Balances of ether: %O', balancesOfEther)
+
+      const balancesOfTokens = await Promise.all(balanceOfTokensPromises)
+      logger.debug('Balances of tokens: %O', balancesOfTokens)
+
+      const balancesOfTokensWithAddress = accountAddresses.map((account, index) => {
+        return {
+          account,
+          name: this._tokensByAccount[accountKeys[index]].name,
+          balancesInfo: balancesOfTokens[index]
+        }
+      })
+      logger.debug('%O', balancesOfTokensWithAddress)
+
+      // Check if the account has ETHER over the reserve amount
+      const balancesOfEtherPromises = balancesOfEther.reduce((balancePromises, balance, index) => {
+        const weiReserveAmount = numberUtil.toWei(ETHER_RESERVE_AMOUNT)
+        if (balance > weiReserveAmount) {
+          const amount = balance.sub(weiReserveAmount)
+          logger.debug('I have to deposit %d ether for account %s', numberUtil.fromWei(amount), accountAddresses[index])
```

Comment:
I would separate clearly were I get the data, and were I execute the transactions.

First we get the `undepositedAmounts` per token.

Then we write using info logger

Then we execute all of the transactions (using the `undepositedAmounts`).